FrEIA/modules/splines/binned.py

```diff
@@ -161,7 +170,16 @@ def constrain_parameters(self, parameters: Dict[str, torch.Tensor]) -> Dict[str,
 
             parameters["widths"] = self.min_bin_sizes[0] + F.softplus(parameters["widths"] + xshift)
             parameters["heights"] = self.min_bin_sizes[1] + F.softplus(parameters["heights"] + yshift)
-            
+
+            domain_width = torch.sum(parameters["widths"], dim=-1, keepdim=True)
+            domain_height = torch.sum(parameters["heights"], dim=-1, keepdim=True)
+            width_resize = self.clamp_domain(domain_width) / domain_width
+            height_resize = self.clamp_domain(domain_height) / domain_height
+
+            parameters["widths"] = parameters["widths"] * width_resize
+            parameters["heights"] = parameters["heights"] * height_resize
+            parameters["left"] = parameters["left"] * width_resize
+            parameters["bottom"] = parameters["bottom"] * height_resize
```

Comment:
I am not exactly sure about the internal behavior of the splines, maybe @LarsKue can comment on this :)

Reply:
Looks good to me. Did log scaling the domain not work @RussellALA ?

Reply:
> Looks good to me. Did log scaling the domain not work @RussellALA ?
My test run ended in NaN. I might try a few more experiments, but it seems to not address the stability issues